pkg/cassandra/management.go

```diff
@@ -0,0 +1,126 @@
+package cassandra
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	cassdcapi "github.com/k8ssandra/cass-operator/operator/pkg/apis/cassandra/v1beta1"
+	"github.com/k8ssandra/cass-operator/operator/pkg/httphelper"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+type ManagementApiFacade interface {
+	CreateKeyspace(
+		ctx context.Context,
+		dc *cassdcapi.CassandraDatacenter,
+		remoteClient client.Client,
+		keyspaceName string,
+		replication map[string]int,
+		logger logr.Logger,
+	) error
+}
+
+type defaultManagementApiFacade struct {
+}
+
+func NewManagementApiFacade() ManagementApiFacade {
+	return &defaultManagementApiFacade{}
+}
+
+func (r *defaultManagementApiFacade) CreateKeyspace(
+	ctx context.Context,
+	dc *cassdcapi.CassandraDatacenter,
+	remoteClient client.Client,
+	keyspaceName string,
+	replication map[string]int,
+	logger logr.Logger,
+) error {
+	if managementApiClient, err := r.createManagementApiClient(ctx, dc, remoteClient, logger); err != nil {
```

Comment:
@jsanda do you know if it's possible to store this client as a struct field and reuse it? Or is it a disposable client?

Reply:
I think it makes sense to store it in some some of request scope variable  (conceptually speaking since there is no such scope in Go). We shouldn't store it across calls to the Reconciler though.